### DISCUSSION TITLE
Lightning

### DISCUSSION THREAD

**Discussion by Bendimester23** [Tnze/go-mc#162] [Ideas]
I know that implementing a full light engine is hard, but a temporary one should be created which just places max skylight in every non solid block

**Comment by maxsupermanhd** [comment]
I think it will be not that hard to port starlight's implementation to go-mc, they have algorithm described. Even though it is kinda recursive and from first glance should not be fast it is for sure better than nothing. https://github.com/PaperMC/Starlight/blob/fabric/TECHNICAL_DETAILS.md

I would implement one at some point when I will work on my archive-server, if you want I can make it a separate package...

**Comment by Bendimester23** [comment]
I understand how starlight works, but i don't know how minecraft handles lights in packets

**Reply by maxsupermanhd** [reply]
Me either, will have to figure that out

**Comment by Tnze** [comment]
We have static skylight now already. Can be read from saves. :)
